pkg/registry/apis/provisioning/repository/github.go

```diff
@@ -45,18 +43,29 @@ func NewGitHub(
 	ctx context.Context,
 	config *provisioning.Repository,
 	factory pgh.ClientFactory,
-	secrets SecretsService,
+	secrets *secrets.Service,
 	webhookURL string,
-) *githubRepository {
+) (*githubRepository, error) {
 	owner, repo, _ := parseOwnerRepo(config.Spec.GitHub.URL)
+	token, encrypted := config.Spec.GitHub.EncryptedToken, true
```

Comment:
[style] Maybe I am wrong but I think it logic could be rewritten to: 

(if token is not already decrypted, take it from the encrypted and decrypt)

```go
	token := config.Spec.GitHub.Token
	if len(token) == 0 {
		var err error
		token, err = secrets.Decrypt(ctx, config.Spec.GitHub.EncryptedToken)
		if err != nil {
			return nil, err
		}
	}
```